js/src/contract/item.ts

```diff
@@ -421,869 +42,283 @@ export class ItemProgram {
-      });
-
-    const instructions = [],
-      signers = [];
+    args: Omit<ItemInstruction.BeginItemActivationArgs, "itemClass">,
+    accounts: ItemInstruction.BeginItemActivationAccounts,
+    additionalArgs: ItemInstruction.BeginItemActivationAdditionalArgs = {},
+    options?: SendOptions
+  ): Promise<{
+    txid: string;
+    slot: number;
+  }> {
+    const signers = [];
     const itemTransferAuthority =
       accounts.itemTransferAuthority || web3.Keypair.generate();
-
     if (
+      accounts.itemAccount &&
       accounts.itemAccount.equals(
         (
           await getAtaForMint(
             accounts.itemMint,
-            (this.program.provider as AnchorProvider).wallet.publicKey
+            (this.client.provider as AnchorProvider).wallet.publicKey
           )
         )[0]
       )
     ) {
       signers.push(itemTransferAuthority);
-
-      instructions.push(
-        Token.createApproveInstruction(
-          TOKEN_PROGRAM_ID,
-          accounts.itemAccount,
-          itemTransferAuthority.publicKey,
-          (this.program.provider as AnchorProvider).wallet.publicKey,
-          [],
-          args.amount.toNumber()
-        )
-      );
     }
-
-    const itemKey = (await getItemPDA(accounts.itemMint, args.index))[0];
-
     const itemClass = await this.fetchItemClass(
       args.itemClassMint,
       args.classIndex
     );
-
-    instructions.push(
-      this.program.methods
-        .beginItemActivation(args)
-        .accounts({
-          itemClass: itemClassKey,
-          itemMint: accounts.itemMint,
-          item: itemKey,
-          itemAccount: accounts.itemAccount,
-          itemTransferAuthority: itemTransferAuthority.publicKey,
-          itemActivationMarker,
-          payer: (this.program.provider as AnchorProvider).wallet.publicKey,
-          playerProgram: PLAYER_ID,
-          systemProgram: SystemProgram.programId,
-          tokenProgram: TOKEN_PROGRAM_ID,
-          rent: web3.SYSVAR_RENT_PUBKEY,
-          clock: web3.SYSVAR_CLOCK_PUBKEY,
-          validationProgram: itemClass.object.itemClassData.config.usages[
-            args.usageIndex
-          ].validation
-            ? new web3.PublicKey(
-                itemClass.object.itemClassData.config.usages[
-                  args.usageIndex
-                ].validation.key
-              )
-            : SystemProgram.programId,
-        })
-        .remainingAccounts(remainingAccounts)
-        .instruction()
-    );
-
-    if (
-      accounts.itemAccount.equals(
-        (
-          await getAtaForMint(
-            accounts.itemMint,
-            (this.program.provider as AnchorProvider).wallet.publicKey
-          )
-        )[0]
-      )
-    ) {
-      instructions.push(
-        Token.createRevokeInstruction(
-          TOKEN_PROGRAM_ID,
-          accounts.itemAccount,
-          (this.program.provider as AnchorProvider).wallet.publicKey,
-          []
-        )
+    if (!itemClass) {
+      throw new Error(
```

Comment:
If you extend the `Error` class as `ItemClassNotFoundError` and throw that, the clients of the library can potentially handle this error programmatically.

Reply:
Great idea - done